mqtt_exporter/main.py

```diff
@@ -48,12 +48,18 @@ def expose_metrics(client, userdata, msg):  # pylint: disable=W0613
         return
 
     for metric, value in payload.items():
-        # we only expose numeric values
-        try:
-            metric_value = float(value)
-        except (ValueError, TypeError):
-            LOG.debug("Failed to convert %s: %s", metric, value)
-            continue
+        if not isinstance(value, (int, float, str, bytes)):
+            continue  # Value is not parsable
+        # we only expose numeric values and ON/OFF as 1/0
+        state_values = {"ON": 1, "OFF": 0}
+        if str(value).upper() in state_values:
+            metric_value = state_values[str(value).upper()]
```

Comment:
result of str(value).upper() should be stored to avoid code duplication.
OPT: only do the test if value is a string (to avoid casting int/float to str)

when it will be in a function: return state_values[..]  > would avoid the "else" below

Reply:
also: do a return anyway if value is a string.

It would permit to remove the try except and do only return float(value) in line 59

Reply:
done